From 1399a98b9802b5aa4e6a88845345a1c1a17bf68d Mon Sep 17 00:00:00 2001
From: Jose Dapena Paz <jdapena@igalia.com>
Date: Tue, 28 Jun 2022 09:00:20 +0200
Subject: libstdc++: std::vector cannot have const members in
 AccountSelectionBubbleView.
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Implementation of std::vector in libstdc++ does not allow const
members.

Fixes the compilation error:
    /usr/include/c++/11/bits/stl_vector.h: In instantiation of ‘class std::vector<const content::IdentityRequestAccount>’:
    ../../chrome/browser/ui/views/webid/account_selection_bubble_view.h:131:60:   required from here
    /usr/include/c++/11/bits/stl_vector.h:401:66: error: static assertion failed: std::vector must have a non-const, non-volatile value_type
      401 |       static_assert(is_same<typename remove_cv<_Tp>::type, _Tp>::value,
          |                                                                  ^~~~~

Bug: 957519
Change-Id: I5bd822243c77526a31db8f152de0326365705522
---
 chrome/browser/ui/views/webid/account_selection_bubble_view.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/chrome/browser/ui/views/webid/account_selection_bubble_view.h b/chrome/browser/ui/views/webid/account_selection_bubble_view.h
index ff851ec50bd38..be537b896462a 100644
--- a/chrome/browser/ui/views/webid/account_selection_bubble_view.h
+++ b/chrome/browser/ui/views/webid/account_selection_bubble_view.h
@@ -128,7 +128,7 @@ class AccountSelectionBubbleView : public views::BubbleDialogDelegateView {
 
   // The list of accounts to select from. Not updated when the user selects an
   // account and navigates to the privacy policy / terms of service page.
-  const std::vector<const content::IdentityRequestAccount> account_list_;
+  const std::vector<content::IdentityRequestAccount> account_list_;
 
   // The TabStripModel of the current browser. We need this in order to show the
   // privacy policy and terms of service urls when the user clicks on the links.
-- 
2.34.1

